docs/reference/query-dsl/has-child-query.asciidoc

```diff
@@ -1,135 +1,158 @@
-`none` and yields the same behaviour as in previous versions. If the
-score mode is set to another value than `none`, the scores of all the
-matching child documents are aggregated into the associated parent
-documents. The score type can be specified with the `score_mode` field
-inside the `has_child` query:
+[[has-child-query-ex-query]]
+===== Example query
 
 [source,js]
---------------------------------------------------
+----
 GET /_search
 {
     "query": {
         "has_child" : {
-            "type" : "blog_tag",
-            "score_mode" : "min",
+            "type" : "child",
             "query" : {
-                "term" : {
-                    "tag" : "something"
-                }
-            }
+                "match_all" : {}
+            },
+            "max_children": 10,
+            "min_children": 2,
+            "score_mode" : "min"
         }
     }
 }
---------------------------------------------------
+----
 // CONSOLE
 
-[float]
-[[min-max-children]]
-==== Min/Max Children
+[[has-child-top-level-params]]
+==== Top-level parameters for `has_child`
 
-The `has_child` query allows you to specify that a minimum and/or maximum
-number of children are required to match for the parent doc to be considered
-a match:
+`type`::
+(string) Required. Name of the child relationship mapped for the
+<<parent-join,join>> field.
 
-[source,js]
---------------------------------------------------
-GET /_search
-{
-    "query": {
-        "has_child" : {
-            "type" : "blog_tag",
-            "score_mode" : "min",
-            "min_children": 2, <1>
-            "max_children": 10, <1>
-            "query" : {
-                "term" : {
-                    "tag" : "something"
-                }
-            }
-        }
-    }
-}
---------------------------------------------------
-// CONSOLE
-<1> Both `min_children` and `max_children` are optional.
+`query`::
+(query object) Required. Query you wish to run on child documents of the `type`
+field. If a child document matches the search, the query returns the parent
+document.
+
+`ignore_unmapped`::
++
+--
+(boolean) Optional. Indicates whether to ignore an unmapped `type` and not return
+any documents instead of an error. Defaults to `false`.
 
-The  `min_children` and `max_children` parameters can be combined with
-the `score_mode` parameter.
+If `false`, {es} returns an error if the `type` is unmapped.
 
-[float]
-==== Ignore Unmapped
+You can use this parameter to query multiple indices that may not contain the
+`type`.
+--
 
-When set to `true` the `ignore_unmapped` option will ignore an unmapped `type`
-and will not match any documents for this query. This can be useful when
-querying multiple indexes which might have different mappings. When set to
-`false` (the default value) the query will throw an exception if the `type`
-is not mapped.
+`max_children`::
+(integer) Optional. Maximum number of child documents allowed for returned parent
```

Comment:
Reading this I was wondering if "child documents" here means child documents that match the query condition. I might have missed that part, but maybe its worth clarifying.

Reply:
Clarified with 6f4668e